routes/routes_impl.py

```diff
@@ -615,6 +622,45 @@ def bathymetry_impl(projection: str, zoom: int, x: int, y: int):
         img = plotting.tile.bathymetry(projection, x, y, zoom, {})
         return _cache_and_send_img(img, f)
 
+def mbt_impl(projection: str, tiletype: str, zoom: int, x: int, y: int):
+  """
+       Serves mbt files
+  """
+  cache_dir = current_app.config['CACHE_DIR']
+  shape_file_dir = current_app.config['SHAPE_FILE_DIR']
+  directory = str(os.path.join(cache_dir, request.path[1:]))
+  # This is a funky way of splitting the string to only include characters up to the 10th instance of '/'
+  basedir = "/".join(directory.split("/", 10)[:10])
+
+  # Send blank tile if conditions aren't met
+  if (zoom < 7) or (projection != "EPSG:3857"):
+    return send_file(shape_file_dir + "/blank.mbt")
+
+# Send file if cached or select data in SQLite file
+  if os.path.isfile(directory):
+    return send_file(directory)
+  else:
+    y = (2**zoom-1) - y
+    connection = sqlite3.connect(shape_file_dir + "/{}.mbtiles".format(tiletype))
```

Comment:
Is it possible `.connect` may throw an exception? Should we check for a failed connection?

Reply:
I don't think there's anyway outside of having to check if the mbtiles files exist before each mbt file can be sent. There's no such check for old topo files (they're just plonked in a `with` block), so I didn't see any reason to add one here. The only way a failed connection could occur is just that the files don't exist in the place they should, which means the navigator isn't installed correctly.